### PR TITLE
🌱 cron: support reading prefix from file for controller input files (7/n)

### PR DESCRIPTION
**Pull request by spencerschrock**
#### What kind of change does this PR introduce?
Part of a series of refactors to generalize the cron infrastructure to enable re-use by the [ossf criticality score project](https://github.com/ossf/criticality_score)

- [X] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
The controller supports a `prefix` which is used to limit which files are used as input files within a bucket e.g. a specific file or directory. The value can be set via the config file or an environment variable (`SCORECARD_INPUT_BUCKET_PREFIX`)

#### What is the new behavior (if this is a feature change)?**
* The `input-bucket` group of prefixes has been grouped and moved under `additional-params`.
  * This will change the corresponding env variables, but for now both are supported.
     * `SCORECARD_INPUT_BUCKET_URL` -> `INPUT_BUCKET_URL`
     * `SCORECARD_INPUT_BUCKET_PREFIX` -> `INPUT_BUCKET_PREFIX`
* Introduction of `prefix-file` (corresponding to env variable of `INPUT_BUCKET_PREFIX_FILE`) whose contents are used as a prefix if `prefix` is not set. This allows for easier changes in situations where the prefix changes frequently e.g. always using the most recent folder in a bucket.

- [X] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
NONE

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

**Comment by spencerschrock** [comment]
@calebbrown FYI

File: cron/internal/controller/bucket.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	s := string(b)
+	return strings.TrimSpace(s), nil
+}
+
+func bucketFiles(ctx context.Context) data.Iterator {
+	var iters []data.Iterator
+
+	bucket, err := config.GetInputBucketURL()
+	if err != nil {
+		panic(err)
```
Comment (naveensrinivasan) [inline cron/internal/controller/bucket.go:63]: Can we avoid panics? Is there a specific reason for not returning errors?

This makes it harder to write tests.

Comment (naveensrinivasan) [reply cron/internal/controller/bucket.go:63]: This comment is applicable across panics in this func.

Comment (naveensrinivasan) [reply cron/internal/controller/bucket.go:63]: Thanks. Appreciate it!

**Review by naveensrinivasan — APPROVED** [review]
Expect the panics; everything else is good. Thanks

**Comment by calebbrown** [comment]
lgtm

**Pull request merged**